api/src/main/java/au/org/aodn/nrmn/restapi/validation/process/ValidationProcess.java

```diff
@@ -1,117 +1,525 @@
+            if (rows.stream().mapToInt(row -> row.getMeasureJson().getOrDefault(measureIndex, 0)).sum() < 50) {
+                columnNames.add(Integer.toString(measureIndex));
+            }
+        List<Long> rowIds = rows.stream().map(r -> r.getId()).collect(Collectors.toList());
+        return columnNames.size() > 0 ? new ValidationError(ValidationCategory.SPAN, ValidationLevel.BLOCKING, "Quadrats do not sum to at least 50 in transect " + transect, rowIds, columnNames) : null;
+    }
+
+    private ValidationError validateSurveyM1M2(List<StagedRowFormatted> surveyRows) {
+        if (surveyRows.stream().filter(r -> Arrays.asList(1, 2).contains(r.getMethod())).collect(Collectors.groupingBy(StagedRowFormatted::getMethod)).size() < 2)
+            return new ValidationError(ValidationCategory.DATA, ValidationLevel.BLOCKING, surveyRows.get(0).getSurveyNum() + ": M1 or M2 missing", surveyRows.stream().map(r -> r.getId()).collect(Collectors.toList()), Arrays.asList("method"));
+        return null;
+    }
+
+    private ValidationError validateSurveyGroup(Integer depth, List<StagedRowFormatted> surveyRows) {
+        if (surveyRows.stream().filter(r -> Arrays.asList(1, 2, 3, 4).contains(r.getSurveyNum())).collect(Collectors.groupingBy(StagedRowFormatted::getSurveyNum)).size() < 4)
+            return new ValidationError(ValidationCategory.SPAN, ValidationLevel.BLOCKING, surveyRows.get(0).getSurveyName() + ": Survey incomplete", surveyRows.stream().map(r -> r.getId()).collect(Collectors.toList()), Arrays.asList("depth"));
+        return null;
+    }
+
+    private Boolean validateATRCM3457(List<StagedRowFormatted> surveyRows) {
+        return surveyRows.stream().filter(r -> Arrays.asList(3, 4, 5, 7).contains(r.getMethod())).count() > 0;
+    }
+
+    public Collection<ValidationError> checkSurveys(String programName, Boolean isExtended, List<StagedRowFormatted> rows) {
+        Set<ValidationError> res = new HashSet<ValidationError>();
+        ValidationResultSet results = new ValidationResultSet();
+
+        Map<Integer, List<StagedRowFormatted>> surveyMap = rows.stream().filter(row -> row.getSite() != null && row.getDate() != null && row.getDepth() != null).collect(Collectors.groupingBy(StagedRowFormatted::getDepth));
+        Map<String, List<StagedRowFormatted>> method3SurveyMap = rows.stream().filter(row -> row.getMethod().equals(3)).collect(Collectors.groupingBy(StagedRowFormatted::getTransectName));
+
+        for (Map.Entry<Integer, List<StagedRowFormatted>> survey : surveyMap.entrySet()) {
+            List<StagedRowFormatted> surveyRows = survey.getValue();
+
+            // VALIDATE: Survey Group Complete
+            res.add(validateSurveyGroup(survey.getKey(), surveyRows));
+
+            // VALIDATE: Is Existing Survey
+            res.add(validateSurveyIsNew(surveyRows.get(0)));
+
+            // VALIDATE: Has M1 M2
+            if (programName == "RLS" || !validateATRCM3457(surveyRows))
```

Comment:
String comparison, so it should be `programName.equals("RLS")`

Reply:
Fixed.